filters/filter.go

```diff
@@ -56,11 +56,32 @@ type Filter interface {
 }
 
 // NewFilter supports startsWith, endsWith, contains and a single wildcard
-// TODO(martinm): add rest of glob matching support and negation
+// along with negation
+// TODO(martinm): add rest of glob matching support
 func NewFilter(pattern string) (Filter, error) {
-	idx := strings.Index(pattern, WildcardChar)
+	nIdx := strings.Index(pattern, negationChar)
+	if nIdx != 0 {
```

Comment:
Yeah folks shouldn't be using it, but unfortunately, we don't drop metrics with "!" chars in it today, so we may have a legacy support issue (metrics are allowed, but you can't create rules for them). We'll need to update the collector package to drop all non-allowed chars first.